### PR TITLE
Fall back to checking out HEAD

### PR DESCRIPTION
**Pull request by fkorotkov**
In case merge ref is not available. These edge cases were cauhgt by integration tests in Cirrus CI:

1. Cloning a PR with conflicts.
2. Cloning a closed PR.

Related to #247 and #249

### CODE REVIEW

**Comment by fkorotkov** [comment]
Tested locally by building a `v1.89.0` version and running Cirrus CI's integration tests locally.

File: internal/executor/executor.go [hunk 1]
```diff
@@ -604,6 +604,12 @@ func (executor *Executor) CloneRepository(
 			fetchOptions.Depth = clone_depth + 1
 		}
 		err = repo.FetchContext(ctx, fetchOptions)
+		if err != nil && strings.Contains(err.Error(), "couldn't find remote ref") {
+			logUploader.Write([]byte("\nFailed to fetch merge ref! PR might not be mergable. Falling back to head ref..."))
+			headRefSpec := fmt.Sprintf("+refs/pull/%s/head:refs/remotes/origin/pull/%[1]s", pr_number)
+			fetchOptions.RefSpecs = []config.RefSpec{config.RefSpec(headRefSpec)}
+			err = repo.Fetch(fetchOptions)
```
Comment (edigaryev) [inline internal/executor/executor.go:611]: Shouldn't we use `FetchContext()` here too?

Comment (fkorotkov) [reply internal/executor/executor.go:611]: I copied the logic from below retry logic. 🤔 Will check if we need to change it in both places.

Comment (fkorotkov) [reply internal/executor/executor.go]: Fixed in [61b473b](https://github.com/cirruslabs/cirrus-ci-agent/pull/250/commits/61b473bd6581783a2c129fba7665c281834ff2ee)

**Review by edigaryev — APPROVED** [review]
